File: python3-bottle/tokenize.py
def tokenize(expression):
    inprogress = ''

    for c in expression:
        if c.isspace():
            continue
        if c.isdigit():
            inprogress += c
        elif c.isalpha():
            raise ValueError
        else:
            if inprogress:
                yield inprogress
                inprogress = ''
            yield c
    if inprogress:
        yield inprogress

File: cxx11-crow/tokenize.cxx
#include <cctype>
#include <list>
#include <string>
#include <iostream>

using namespace std;

namespace infixcalc {
  list<string> tokenize(string expression) {
    string inprogress = "";
    list<string> result;

    for (char c : expression) {
      if (isspace(c)) {
        continue;
      }
      if (isdigit(c)) {
        inprogress += c;
      } else if (isalpha(c)) {
        throw;
      } else {
        if (inprogress.length() > 0) {
          result.push_back(inprogress);
          inprogress = "";
        }
        string charstr(1, c);
        result.push_back(charstr);
      }
    }
      if (inprogress.length() > 0) {
        result.push_back(inprogress);
      }
      return result;
  }
};

// int main() {
//   auto result = infixcalc::tokenize("1 + (2 * 3)");
//   for (auto token: result) {
//     cout << '"' << token << '"' << ' ';
//   }
//   cout << endl;
// }
